.github/workflows/test.yml

```
@@ -1,19 +1,16 @@
 name: Test
 
-on: [push]
+on: [ push ]
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
       - name: Install minimal stable
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-      - uses: actions/checkout@v3
+        uses: dtolnay/rust-toolchain@stable
+      - uses: actions/checkout@v4
       - name: Install dependencies
-        run: sudo apt-get install protobuf-compiler
+        run: sudo apt update; sudo apt upgrade -y;sudo apt-get install protobuf-compiler
```

Since this is non-interactive, we probably want `apt-get`.

```suggestion
        run: sudo apt-get update; sudo apt-get upgrade -y; sudo apt-get install protobuf-compiler
```